### PR TITLE
Added JwtToken class

### PR DESCRIPTION
**Pull request by CSchulz**
Replaced any with JwtToken in JwtHelper.decodeToken()

Based on https://tools.ietf.org/html/rfc7519#section-4.1.1

### CODE REVIEW

**Comment by CSchulz** [comment]
@chenkie @escardin Any comments?

**Comment by escardin** [comment]
What value is it providing over a raw object? How does deal with the potential assumption by users that just because a field is specified, doesn't mean it's in the token?

I do like the idea of a token being a little smarter, but I actually despise getters and setters.

**Comment by CSchulz** [comment]
In general it would be possible to throw an error and / or define the return value as undefined too.

I don't like the getters too, but they provide autocomplete of properties based on the RFC.

Perhaps it is possible to implement a Proxy for a dynamic getter implemention.

**Comment by escardin** [comment]
How about a typescript interface?

**Comment by CSchulz** [comment]
I think it would be also okay. I would refactor all stuff again. :laughing:

**Comment by chenkie** [comment]
anything more to do here or are we good to merge?

**Comment by CSchulz** [comment]
I had no time, I will do it the next days hopefully.

File: angular2-jwt.ts [hunk 1]
```diff
@@ -53,6 +53,68 @@ const AuthConfigDefaults: IAuthConfig = {
     noTokenScheme: false
 };
 
+export class JwtToken {
```
Comment (9orky) [inline angular2-jwt.ts]: Using name JwtToken is like saying RAM Memory ;-) How about "Jwt" only?

**Comment by 9orky** [comment]
How about implementing .toString() method so one can store Jwt string without using getter? :)

File: angular2-jwt.ts (hunk 1 above)
Comment (CSchulz) [inline angular2-jwt.ts]: Done. 😜

**Comment by CSchulz** [comment]
@escardin what about a class with string index, all parameters assigned and having a toString() method returning the token string?

**Comment by escardin** [comment]
toString is for debugging and display. It would not be appropriate to use it to get the token. Have a token.token or token.rawToken getter. 

Somewhat related article & comments: https://news.ycombinator.com/item?id=15046223

**Pull request closed without merging**